src/librustc_resolve/lib.rs

```diff
@@ -4796,10 +4796,18 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                 err.span_suggestion_with_applicability(
                     binding.span,
                     rename_msg,
-                    if snippet.ends_with(';') {
-                        format!("{} as {};", &snippet[..snippet.len() - 1], suggested_name)
+                    if snippet.contains(" as ") {
```

Comment:
This will only work with spaces, but these could be written in any of the following ways:

```
use foo as bar;
use
foo
as
bar;
use\tfoo\tas\tbar;
use foo\tas bar;
use foo as\tbar;
use foo
as\tbar;
use foo\tas
bar;
```

:)

You probably want to use https://doc.rust-lang.org/std/primitive.str.html#method.split_whitespace

Reply:
thanks, didn't think of that as I'm so used to rustfmt